### DISCUSSION TITLE
Group by Sum change signage based on condition

### DISCUSSION THREAD

**Discussion by thesobercoder** [prisma/prisma#5822] [Q&A] [answered]
In a typical SQL query, I would be able to do something similar like 

```
SELECT SUM(CASE WHEN A = 'X' THEN QUANTITY * -1 ELSE QUANTITY)
FROM TABLE
GROUP BY B
```

where I'm able to either make the quantity negative or positive based on whether it's a buy or a sell. How can I achieve the same with prisma group by?

**Comment by matthewmueller** [comment, marked as answer]
Hey @thesobercoder! 

Unfortunately, rich SQL expressions like `CASE WHEN A = 'X' THEN QUANTITY * -1 ELSE QUANTITY` are not possible yet. For the time being, I'd suggest computing this expression in your application code using the [groupBy API](https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by-preview). 

Something like _(untested)_:

```ts
const groupBs = await prisma.TABLE.groupBy({
  by: ["B", "A"],
  sum: {
    QUANTITY: true
  }
})
const result = groupBs.map(group => group.A == 'X' ? group.sum.QUANTITY * -1 : group.sum.QUANTITY)
```



Alternatively, the [$queryRaw API](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access/#queryraw) is the way to go.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
